prisma-fmt/src/validate.rs

```diff
@@ -0,0 +1,97 @@
+use serde_json::json;
+
+// this mirrors user_facing_errors::common::SchemaParserError
+pub static SCHEMA_PARSER_ERROR_CODE: &str = "P1012";
+
+pub(crate) fn run(input_schema: &str) -> Result<(), String> {
+    let validate_schema = psl::validate(input_schema.into());
+    let diagnostics = &validate_schema.diagnostics;
+
+    if !diagnostics.has_errors() {
+        return Ok(());
+    }
+
+    // always colorise output regardless of the environment
+    colored::control::set_override(true);
+
+    use std::fmt::Write as _;
```

Comment:
Could this `use` move at the top of the function or the module?

Reply:
Ok, originally it was declared right above the `write!` macro in the query-engine's code, that's why I had kept it local to it in `prisma-fmt`

Reply:
Makes sense. Lots of bad code in the query engine :p